### PR TITLE
Fix odo describe not listing all the ports after first url is created

### LINKED ISSUE #3201: odo describe for a component does not lists down all the ports if a url is already created
/kind bug

## What versions of software are you using?
odo 1.2.1

**Operating System:** 

**Output of `odo version`:**
```
bash-3.2$ "/Users/mohitsuman/Projects/vscode-openshift-tools/out/tools/darwin/odo" version
odo v1.2.1 (b7b7c2fac)
Server: https://api.crc.testing:6443
Kubernetes: v1.14.6+b294fe5
```

# How did you run odo exactly?
1. Create a URL from IDE and select the port(if the component has multiple ports)
2. Create a new URL second time, here the `odo describe -ojson` output for a component doesn't list down all the ports available for the component, the ports object just has the entry of the port selected in Step 1.

## Actual behavior
Step 1:  For a given component: `odo describe -ojson` lists all ports available for the component
Screenshot 2020-05-18 at 6 07 45 PM

Step 2: For the same component: `odo describe -ojson` only lists the port selected in Step1. It does not list all ports.
Screenshot 2020-05-18 at 6 12 38 PM

## Expected behavior

Whenever we run `odo describe -ojson` for a component , all the ports available to a component should be listed down.

Repo used for reference: Java component: https://github.com/spring-projects/spring-petclinic

### PR DESCRIPTION
**Pull request by dgolovin**
This PR fixes #3201 and gets back original behaviour.

Signed-off-by: Denis Golovin dgolovin@redhat.com

### CODE REVIEW

**Comment by dharmit** [comment]
This works to the end of fixing #3201 but it breaks elsewhere (read the comments before each command):
```sh
# Create a URL for port 8443
$ odo url create url1 --port 8443
 ✓  URL url1 created for component: java-spring-petcl-kzqa

To apply the URL configuration changes, please use `odo push`

# url list shows the port value correctly
$ odo url list
Found the following URLs for component java-spring-petcl-kzqa in application app:
NAME     STATE          URL     PORT     SECURE
url1     Not Pushed     ://     8443     false
There are local changes. Please run 'odo push'.

# But odo describe shows that url1 will expose port 8080 instead of 8443
$ odo describe
Component Name: java-spring-petcl-kzqa
Type: java
Source: file://./
Environment Variables:
 · DEBUG_PORT=5858
URLs:
 · URL named url1 will be exposed via 8080/TCP
```
However, `odo describe -o json` shows the port-url mapping correctly:
```yaml
"items": [
    {
        "kind": "url",
        "apiVersion": "odo.dev/v1alpha1",
        "metadata": {
            "name": "url1",
            "creationTimestamp": null
        },
        "spec": {
            "port": 8443,
            "secure": false
        },
        "status": {
            "state": "Not Pushed"
        }
    }
]

```
Since json output is correct, I think we can merge this PR and work on `odo describe` output separately.

@dgolovin thanks for the fix and, once again, sorry that my commit broke stuff in json output. Things messed up in trying to fix two things in that PR.

/approve

**Comment by dharmit** [comment]
> Since json output is correct, I think we can merge this PR and work on `odo describe` output separately.

#3229

**Comment by kadel** [comment]
@dgolovin I just pushed commit to your branch to make tests pass

File: tests/integration/component.go [hunk 1]
```diff
@@ -249,7 +249,7 @@ func componentTests(args ...string) {
...
+			expected, err := helper.Unindented(`{"kind": "Component","apiVersion": "odo.dev/v1alpha1","metadata": {"name": "cmp-git","namespace": "` + project + `","creationTimestamp": null},"spec":{"app": "testing","type":"nodejs","source": "https://github.com/openshift/nodejs-ex","sourceType": "git","urls": {"kind": "List", "apiVersion": "odo.dev/v1alpha1", "metadata": {}, "items": [{"kind": "url", "apiVersion": "odo.dev/v1alpha1", "metadata": {"name": "url-1", "creationTimestamp": null}, "spec": {"port": 8080, "secure": false}, "status": {"state": "Not Pushed"}}, {"kind": "url", "apiVersion": "odo.dev/v1alpha1", "metadata": {"name": "url-2", "creationTimestamp": null}, "spec": {"port": 8080, "secure": false}, "status": {"state": "Not Pushed"}}]},"storages": {"kind": "List", "apiVersion": "odo.dev/v1alpha1", "metadata": {}, "items": [{"kind": "storage", "apiVersion": "odo.dev/v1alpha1", "metadata": {"name": "storage-1", "creationTimestamp": null}, "spec": {"size": "1Gi", "path": "/data1"}}]},"ports": ["8080/TCP"]},"status": {"state": "Not Pushed"}}`)
```
Comment (amitkrout) [inline tests/integration/component.go:252]: /lgtm

**Comment by dgolovin** [comment]
@dharmit good point, I did not notice that. I was just thinking if I keep initial value for ports everything is going to be fine, but I was wrong.

**Comment by dgolovin** [comment]
@kadel thanks for taking care of tests!

**Comment by dharmit** [comment]
Keeping this PR on hold while I'm fixing #3193 

/hold

**Comment by dharmit** [comment]
> Keeping this PR on hold while I'm fixing #3193
> 
> /hold

Opened PR #3239 to fix this issue.

**Pull request merged**